search-service/src/main/kotlin/com/egm/stellio/search/entity/service/EntityService.kt

```diff
@@ -574,21 +599,20 @@ class EntityService(
     ): Either<APIException, Unit> = either {
         authorizationService.userCanUpdateEntity(entityId, sub.toOption()).bind()
 
-        when (attributeName) {
-            NGSILD_SCOPE_PROPERTY -> scopeService.delete(entityId).bind()
-            else -> {
-                entityAttributeService.checkEntityAndAttributeExistence(
-                    entityId,
-                    attributeName,
-                    datasetId
-                ).bind()
-                entityAttributeService.deleteAttribute(
-                    entityId,
-                    attributeName,
-                    datasetId,
-                    deleteAll
-                ).bind()
-            }
+        if (attributeName == NGSILD_SCOPE_PROPERTY) {
+            scopeService.delete(entityId).bind()
+        } else {
+            entityAttributeService.checkEntityAndAttributeExistence(
+                entityId,
+                attributeName,
+                datasetId
+            ).bind()
+            entityAttributeService.deleteAttribute(
+                entityId,
+                attributeName,
+                datasetId,
+                deleteAll
+            ).bind()
```

Comment:
I think the new implementation is less clear.
I am not sure there is a need to enforce this rule.

Reply:
actually a standard Kotlin convention: https://kotlinlang.org/docs/coding-conventions.html#if-versus-when

but I agree the previous version was a bit more readable

Reply:
In the rest of the PR, the swap from 'when' to 'if' was an improvement. So it makes sense to keep the rule